### DISCUSSION TITLE
cannot receive data on remote server

### DISCUSSION THREAD

**Discussion by 6273parastu** [miguelgrinberg/Flask-SocketIO#1925] [Q&A]
Hi, I developed simple project with flask soket io and try to get real time voice in backend. in localhost it works and receive data but when i run it in remote server it doesn`t work.

```python
@socketio.on('write-audio')
def write_audio(data):
    """Write a chunk of audio from the client."""
    print(type(data))
  

@socketio.on('connect')
def on_connect():
    print('on_connect')
    socketio.start_background_task(target=target_class.background_thread_from_stream, app=current_app._get_current_object())
``` 
```javascript
socket.on('connect', function() {
    console.log('CONNECTION ESTABLISHED !!')
});
soket.emit('write-audio', buffer)
```

**Comment by 6273parastu** [comment]
Request URL: http://192.168.35.2:6300/socket.io/?EIO=4&transport=polling&t=OK5QrnD&sid=y4YTNdrdFiHu-p6rAACa
Request Method: GET
Status Code: 400 BAD REQUEST
"Invalid session"

**Reply by miguelgrinberg** [reply]
It's really impossible for me to tell you what's wrong, but my guess is that this is a problem with how you deployed your application. Have you read the documentation, specifically the section that describes deployment options? I would suggest you start from there.
